Linking from leaderboard to Raw Runs/Predictions

farzaank opened the pull request:
- Initial functionality for linking to raw runs from leaderboard
  - Uses converts display_names in tables into raw names to use for querying raw runs
- Currently works in a hacky way by pre-filling a query in the Predictions page to find related runs
  - For now, I opted to do this as doing this directly within the tooltip would require the leaderboard to load raw runs (correct me if I'm wrong here) and find the related ones which seems inefficient
  - If there's a better way to do this I can implement it, but I think this should suffice for bringing users from leaderboards to the related runs they could be interested in, we just have some additional work in potentially reducing the friction here
  
As usual, feel free to test here: https://farzaank.github.io/helm/#/leaderboard
 
Example:
  
<img width="770" alt="Screenshot 2023-12-15 at 3 14 42 PM" src="https://github.com/stanford-crfm/helm/assets/39839866/0c422bb2-5b79-4cc6-8cd2-f4f4fbc57377">


<img width="873" alt="Screenshot 2023-12-15 at 3 15 13 PM" src="https://github.com/stanford-crfm/helm/assets/39839866/5796ef2d-1eee-4af1-8776-41ee9417cdf4">

farzaank commented on src/helm-frontend/src/components/LeaderboardTables.tsx:
```diff
@@ -37,6 +40,107 @@ export default function LeaderboardTables({
   const [filteredModels, setFilteredModels] =
     useState<string[]>(modelsToFilter);
 
+  const [schema, setSchema] = useState<Schema | undefined>(undefined);
```
For cleanliness I can move this out into a separate component when I refactor the leaderboard next quarte.r

yifanmai commented:
Seems a bit iffy to be honest:

1. The tooltip is not great because it blocks a lot of information and affects copy and pasting text. One alternative is to make the cell itself the link, and then use the `title` HTML attribute for the tooltip (i.e. native browser tooltip rather than an extra React element).
2. This doesn't work on the group-level tables (where the links should be attached to the _rows_ rather than the _cells_.
3. The links to the runs can be incorrect because the regex is wrong in many cases. This is a _pre-existing bug that applied to the old version_. The better way to do this is to have `helm-summarize` output the lists of runs.

I should also re-emphasize that this is a completely new feature not a backport of an existing feature from the old frontend, because (1) the old frontend never supported cell level links and (2) the old frontend had incorrect run regexes in many cases.

I think this requires some design review, so I'd suggest leaving this for now and revisiting it in January.

farzaank commented:
> Seems a bit iffy to be honest:
> 
> 1. The tooltip is not great because it blocks a lot of information and affects copy and pasting text. One alternative is to make the cell itself the link, and then use the `title` HTML attribute for the tooltip (i.e. native browser tooltip rather than an extra React element).
> 2. This doesn't work on the group-level tables (where the links should be attached to the _rows_ rather than the _cells_).
> 3. The links to the runs can be incorrect because the regex is wrong in many cases. This is a _pre-existing bug that applied to the old version_. The better way to do this is to have `helm-summarize` output the lists of runs.
> 
> I should also re-emphasize that this is a completely new feature not a backport of an existing feature from the old frontend, because (1) the old frontend never supported cell level links and (2) the old frontend had incorrect run regexes in many cases.
> 
> I think this requires some design review, so I'd suggest leaving this for now and revisiting it in January.

Addressing 1 and 2 is pretty straightforward (I actually started with implementing your first suggestion) and I've already thought about how to implement 2, but I also noticed 3 was an issue and am not really sure how to fix it. Accordingly, I can wait until we have a way to address it. My only idea was manual inspection of what works and doesn't and hard coding edge cases, but that's likely untenable.

yifanmai commented:
Chatted offline - for this PR, we will get a working UI but use regexes for run names. We will have a follow-up PR for `summarize.py` to produce the correct run names.

farzaank commented:
This addresses only feedback point 1, and 2 & 3 will be addressed in a follow-up that has #2232 and this as a dependency.

I have reverted to the previous implementation of the tooltip (using the native HTML one). I also archived the previous implementation in a separate place in case we want to create a custom tooltip at some point.

The reason I changed to a custom tooltip was that the delay was acting weird in some cases, but after some thought I think it's better than how I implemented the custom tooltip. We could also just have the custom tooltip behave like the native one, but that's probably not worth it at this point so I can revisit later if we get feedback about it.

yifanmai reviewed (changes requested):
The native tooltip looks good.

I tried running this locally and it seems to break for me in a number of ways - the table layout seems broken, and the cells seems to be reverting to linking to the `href` field (which contains broken links that we should eventually get rid of).

![Screenshot 2024-01-17 111455](https://github.com/stanford-crfm/helm/assets/185227/361d2175-03a8-4dc9-8a9c-c4ecd4c269e5)

I'll take a closer look at the code this afternoon.

farzaank commented:
> The native tooltip looks good.
> 
> I tried running this locally and it seems to break for me in a number of ways - the table layout seems broken, and the cells seems to be reverting to linking to the `href` field (which contains broken links that we should eventually get rid of).
> 
> ![Screenshot 2024-01-17 111455](https://private-user-images.githubusercontent.com/185227/297496308-361d2175-03a8-4dc9-8a9c-c4ecd4c269e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1MjA2NjAsIm5iZiI6MTcwNTUyMDM2MCwicGF0aCI6Ii8xODUyMjcvMjk3NDk2MzA4LTM2MWQyMTc1LTAzYTgtNGRjOS04YTljLWM0ZWNkNGMyNjllNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExN1QxOTM5MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNjA5MWJlYWM2Y2VkOGY0Y2MyNDA1ZGYxYzc5MjE5MzM1N2Y1ZWE4MGJkZmY0ZTUwZGRkZWU2NGY5NWI2ZDI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JUjQF5y3bWC3qhWBIksj1ICYH9b5eacdvgPNQFeS-5g)
> 
> I'll take a closer look at the code this afternoon.

That's weird, I think one of my last style changes broke the table and didn't propagate while I was locally testing. I reverted that change and the format is back to working.

W.r.t the broken links, do we just want to not link to regexes yet and fix that in a separate PR? Alternatively I can merge in your changes and address everything in this PR.

yifanmai commented on helm-frontend/src/components/LeaderboardTables.tsx:
```diff
@@ -37,6 +39,63 @@ export default function LeaderboardTables({
+  const getGroupForRunName = (rawGroup: string): string => {
+    const groupSplit = replaceLastHyphen(rawGroup).split("*");
+    const group = groupSplit[0].trim();
+    console.log(group);
+    console.log(schema?.run_groups);
```
Delete this.

farzaank replied:
I'll handle this and #2 before merging, then handle #1 in the subsequent PR.

farzaank replied:
Handled this, but leaving out the second change as it changes the style in an ugly way (making every thing a bright blue link).

yifanmai reviewed (approved):
Looks good!

In the subsequent PR where you use `run_spec_names`, could you also do the following?

1. Remove the dependency on `getSchema` - I think the tooltip can be something generic like "get predictions for aggregate metric value".
2. Push the link down into `RowValue` - it already has some handling for links.
